src/optimize.jl

```diff
@@ -25,18 +25,18 @@ function extract_inner_call!(stmt::Expr, idx, once::Bool=false)
     return nothing
 end
 
-function replace_ssa!(@nospecialize(stmt), ssalookup)
-    isa(stmt, Expr) || return nothing
-    for (i, a) in enumerate(stmt.args)
+function replace_ssa(@nospecialize(stmt), ssalookup)
+    isa(stmt, Expr) || return stmt
+    return Expr(stmt.head, Any[
         if isa(a, SSAValue)
-            stmt.args[i] = SSAValue(ssalookup[a.id])
+            SSAValue(ssalookup[a.id])
         elseif isa(a, NewSSAValue)
-            stmt.args[i] = SSAValue(a.id)
+            SSAValue(a.id)
         else
-            replace_ssa!(a, ssalookup)
+            replace_ssa(a, ssalookup)
         end
-    end
-    return nothing
+        for a in stmt.args
+    ]...)
```

Comment:
Isn't splatting a potentially very long vector considered bad form?

Reply:
This vector should typically be relatively short, since exprs don't usually have that many arguments. There isn't currently a nice way to create an expression like this without splatting unfortunately, but since the constructor is a builtin there shouldn't be any unnecessary specialization, so this should be fine.

Reply:
Couldn't you write this as
```
ex = Expr(stmt.head)
ex.args = Any[
    if isa(a, SSAValue)
        SSAValue(ssalookup[a.id])
    elseif isa(a, NewSSAValue)
        SSAValue(a.id)
    else
         replace_ssa(a, ssalookup)
    end
    for a in stmt.args
]
return ex
```
But yeah, as you said, doesn't matter much ;)